### ISSUE TITLE
Update warnings and announcements

### ISSUE THREAD

**Issue by ChrisRackauckas** [JuliaLang/Pkg3.jl#176]
DiffEq has been using a hack by @Datseris to display an announcement in the DiffEq repos. 

https://github.com/JuliaDiffEq/DiffEqBase.jl/blob/master/deps/build.jl

While it works, it's quite hacked on so there's issues like https://github.com/JuliaDiffEq/DiffEqBase.jl/issues/94 , but I am wondering if there could be a better way to do announcements than "follow the entire repo" that are built into Pkg3. I for one would love to know about all releases of dependencies for example.

**Comment by KristofferC** [comment]
Do you want this metatada to be stored even if the package is deleted? Otherwise, you can just touch a file and look for the presence of it https://github.com/JuliaCI/PkgBenchmark.jl/blob/master/deps/build.jl.

**Comment by Datseris** [comment]
Oh wow. That seems like a much better approach!

**Comment by KristofferC** [comment]
(remember to add the file to `.gitignore`)

**Comment by tpapp** [comment]
It would be great if the solution above could end up as a suggestion in the **documentation** when that is written, so maybe this issue could be kept open as a reminder.

**Comment by KristofferC** [comment]
Since we currently redirect the build log to an output file the workaround described above is no longer effective.

**Comment by Datseris** [comment]
> Since we currently redirect the build log to an output file the workaround described above is no longer effective.

Oh no, seemed like an awesome idea. Do you have any alternative suggestions?

**Comment by KristofferC** [comment]
You could do almost the same thing but instead when loading the package for the first time.

**Comment by Datseris** [comment]
@KristofferC I did this, and it worked like a charm. The update is displayed only once. For reference, this is what I have inside my main `MyModule.jl` file:
```
if !isfile(joinpath(@__DIR__, "update_v2.0.0"))
    printstyled(stdout,
    """
    Update text here.
    """; color = Base.info_color())
    touch("update_v2.0.0")
end
```
However, there is something I don't understand. Where is this file being created? To my understanding it should be at the directory where the package is installed, no? Yet when I go to this directory, I can't seem to find the file:
```
julia> pwd()
"C:\\Users\\datseris\\.julia\\dev\\DynamicalBilliards"

julia> readdir()
14-element Array{String,1}:
 ".git"
 ".gitignore"
 ".travis.yml"
 "appveyor.yml"
 "benchmark"
 "CHANGELOG.md"
 "docs"
 "ISSUE_TEMPLATE.md"
 "LICENSE.md"
 "paper"
 "README.md"
 "REQUIRE"
 "src"
 "test"
```
I wanted to delete it to re-test the update message, but I can't find it :D

**Comment by Datseris** [comment]
Hm, turns it I am not sure whether it worked. I did some changes to my file, but I did not change the file's name.

When `DynamicalBilliards` was re-compiled, the update message was displayed again.

**Comment by fredrikekre** [comment]
Since you used `@__DIR__` from `MyModule.jl` the file will end up in `src`, no?

**Comment by Datseris** [comment]
You are correct. It should be in the `src`. It isn't though. I think for some reason this file may go where the precompiled caches are?... but it is also not in `\.julia\compiled\v0.7\DynamicalBilliards` damn!

So the update message just got displayed again after re-compilation. The file though is not in the main folder, not in the `src` folder, or in the compile folder... 

Could it be that `touch` does *not* create a file?

**Comment by Datseris** [comment]
damn I am such a huge moron... look at the code I pasted. I do `touch(...)` but when searching for the file I Do `isfile(joinpath(@__DIR__, "update_v2.0.0"))` The correct thing is to also do `touch(joinpath(@__DIR__, "update_v2.0.0"))` !!!
